src/dolphin/main.ts

```diff
@@ -5,23 +5,111 @@ import path from "path";
 
 import { fileExists } from "../main/fileExists";
 import { assertDolphinInstallations } from "./downloadDolphin";
+import { GeckoCode, geckoCodeToRaw, removeGeckoCode, saveCodes, setEnabledDisabledFromTCodes } from "./geckoCode";
+import { IniFile } from "./iniFile";
 import {
+  ipc_addGeckoCode,
   ipc_checkDesktopAppDolphin,
   ipc_checkPlayKeyExists,
   ipc_clearDolphinCache,
   ipc_configureDolphin,
+  ipc_convertGeckoToRaw,
+  ipc_deleteGecko,
   ipc_downloadDolphin,
+  ipc_fetchGeckoCodes,
+  ipc_fetchSysInis,
   ipc_importDolphinSettings,
   ipc_launchNetplayDolphin,
   ipc_reinstallDolphin,
   ipc_removePlayKeyFile,
   ipc_storePlayKeyFile,
+  ipc_toggleGeckos,
   ipc_viewSlpReplay,
 } from "./ipc";
 import { dolphinManager } from "./manager";
 import { deletePlayKeyFile, findPlayKey, writePlayKeyFile } from "./playkey";
 import { DolphinLaunchType } from "./types";
-import { findDolphinExecutable, updateBootToCssCode } from "./util";
+import {
+  findDolphinExecutable,
+  findSysFolder,
+  findUserFolder,
+  getUserIni,
+  loadCodes,
+  updateBootToCssCode,
+} from "./util";
+
+ipc_fetchGeckoCodes.main!.handle(async ({ dolphinType, iniName }) => {
+  console.log("fetching gecko codes...");
+  const gCodes = await loadCodes(dolphinType, iniName);
+  const tCodes: GeckoCode[] = [];
```

Comment:
what is the point of this? gCodes is already an array of gecko codes that you could just return. Also why is it called tCodes?

Reply:
tCodes is for truncated gecko codes, the notes and codeLines are empty in tCodes as opposed to gCodes so we don't end up storing codelines/notes (which can be very large) in the state of dolphinSettings

Reply:
gotcha, adding that as a comment would be a good idea.

we can also use the spread operator like so to simplify the code a bit
```ts
const tCode: GeckoCode = {
      ...gcode
      notes: [],
      codeLines: [],
    };
```
or just modify gCode in place and push that into tCodes. Either option is a bit cleaner.